WcaOnRails/app/models/user.rb

```diff
@@ -315,10 +315,10 @@ def avatar_requires_wca_id
 
   after_save :remove_pending_wca_id_claims
   private def remove_pending_wca_id_claims
-    if delegate_status_changed? && !delegate_status
+    if delegate_status_before_last_save && !delegate_status
```

Comment:
Ah, that works too, and is closer to what we used to have. Either way the tests pass. In English, the two sentences are:

1. Did we just change this guy's delegate status and is he currently not a delegate?
2. Was this guy previously a delegate and is he currently not a delegate?

I've changed it from 2) to 1), as you suggested.